lib/generators/rolemodel/css/base/templates/app/javascript/stylesheets/utilities.scss

```diff
@@ -24,70 +24,35 @@
 
 .flex-wrap { flex-wrap: wrap; }
 
-.flex-grow-1 { flex-grow: 1; }
-
-// Layout properties
-// E.G:
-// .flex.layout-row.gap-md
-//   .two-thirds-width.card Item
-//   .card Item
-//   .card Item
+.flex-no-wrap { flex-wrap: nowrap; }
 
-// Normal width
-// |      Item     | | Item | | Item |
-// Mobile
-// | Item |
-// | Item |
-// | Item |
-
-// E.G:
-// .flex.layout-row.gap-md.flex-wrap
-//   .two-thirds-width.card Item
-//   .card Item
-//   .card Item
-
-// Normal width
-// |      Item     | | Item | | Item |
-// Mobile
-// |   Item   | | Item |
-// |        Item       |
-
-.layout-row {
-  & > * {
-    flex: 1;
-
-    &.four-fifths-width { flex-basis: calc(100% * 4/5); }
-    &.three-fifths-width { flex-basis: calc(100% * 3/5); }
-    &.two-fifths-width { flex-basis: calc(100% * 2/5); }
-    &.one-fifth-width { flex-basis: calc(100% * 1/5); }
-
-    &.three-quarters-width { flex-basis: calc(100% * 3/4); }
-    &.one-quarter-width { flex-basis: calc(100% * 1/4); }
-
-    &.two-thirds-width { flex-basis: calc(100% * 2/3); }
-    &.one-third-width { flex-basis: calc(100% * 1/3); }
-
-    &.full-width { flex-basis: 100%; }
-    &.half-width { flex-basis: 50%; }
-  }
-
-  @media screen and (max-width: $breakpoint-sm) {
-    flex-wrap: wrap;
-
-    &:not(.flex-wrap) > * {
-      flex-basis: 100%;
-      width: 100%;
-    }
-  }
-}
+.flex-grow-1 { flex-grow: 1; }
 
 // Gap Properties
-.gap-xxs { gap: var(--space-xxs); }
-.gap-xs { gap: var(--space-xs); }
-.gap-sm { gap: var(--space-xxs); }
-.gap-md { gap: var(--space-md); }
-.gap-lg { gap: var(--space-lg); }
-.gap-xl { gap: var(--space-xl); }
+.gap-xxs {
+  gap: var(--space-xxs);
+  --gap: var(--space-xxs);
```

Comment:
@Jeremy-Walton Why do we need to add the custom properties within the classes here?

Reply:
The reason for that is so that when something under an element with a gap set, it can have access to specifically which gap is set (via the --gap variable). The specific use case is when using `flexible-layout-area` with a gap set and using the width utilities e.g. `one-fifth-width` under it. Under the `flexible-layout-area`, those width utils use `flex-basis` to calculate width which does not take into account any gap set, so I had to make a way that they could include the gap when calculating the width. e.g. `.one-fifth-width { flex-basis: calc(calc(100% * 1/5) - var(--gap)); }`.

If I didn't set the `--gap`, I wouldn't know which spacing variable to subtract out here